### PR TITLE
Added support to provide specify status bar color

### PR DESCRIPTION
**Pull request by adamhrz**
@rztakashi @ZevEisenberg  Based on feedback in #11 you can now set a preferredStatusBarStyle in your messaging view controller and it will animate in with the message and back out (if the underlying status bar is a different color).

### CODE REVIEW

File: Pod/Classes/RZErrorMessagingViewController.m [hunk 1]
```diff
@@ -61,6 +61,14 @@ - (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibB
     return self;
 }
 
+/**
+ *  Set the style of the status bar for the message
+ */
```
Comment (jvisenti) [inline Pod/Classes/RZErrorMessagingViewController.m]: Not sure this method actually need a comment.

File: Pod/Classes/RZMessagingWindow.m [hunk 2]
```diff
@@ -292,28 +292,48 @@ @implementation RZRootMessagingViewController
...
+    UIStatusBarStyle statusBarStyle;
     
-    if ( topViewController == self ) {
-        return [super shouldAutorotate];
+    if ( [(RZMessagingWindow *)self.view.window errorIsBeingPresented] ) {
+        statusBarStyle = [super preferredStatusBarStyle];
     }
     else {
-        return topViewController.shouldAutorotate;
+        statusBarStyle = topViewController.preferredStatusBarStyle;
```
Comment (jvisenti) [inline Pod/Classes/RZMessagingWindow.m]: `preferredStatusBarStyle` isn't a property so shouldn't use dot notation. @ZevEisenberg That's our style right?

File: Pod/Classes/RZMessagingWindow.m [hunk 3]
```diff
@@ -292,28 +292,48 @@ @implementation RZRootMessagingViewController
...
     
-    if ( topViewController == self ) {        
-        return [super supportedInterfaceOrientations];
+    UIViewController *childViewController;
+    if ( [(RZMessagingWindow *)self.view.window errorIsBeingPresented] ) {
+        childViewController = [self.childViewControllers lastObject];
     }
     else {
-        return topViewController.supportedInterfaceOrientations;
+        childViewController = topViewController.childViewControllerForStatusBarStyle;
```
Comment (jvisenti) [inline Pod/Classes/RZMessagingWindow.m]: Same comment about dot notation

File: Pod/Classes/RZMessagingWindow.m [hunk 4]
```diff
@@ -292,28 +292,48 @@ @implementation RZRootMessagingViewController
...
     }
+    return childViewController;
+    
+}
+
+-(BOOL)shouldAutorotate
+{
+    UIViewController *topViewController = [RZRootMessagingViewController topViewController];
+    
+    return topViewController.shouldAutorotate;
```
Comment (jvisenti) [inline Pod/Classes/RZMessagingWindow.m]: dot notation

File: Pod/Classes/RZMessagingWindow.m [hunk 5]
```diff
@@ -292,28 +292,48 @@ @implementation RZRootMessagingViewController
...
+}
+
+- (NSUInteger)supportedInterfaceOrientations
+{
+    UIViewController *topViewController = [RZRootMessagingViewController topViewController];
+    
+    return topViewController.supportedInterfaceOrientations;;
```
Comment (jvisenti) [inline Pod/Classes/RZMessagingWindow.m]: dot notation

File: Pod/Classes/RZMessagingWindow.m (hunk 2 above)
Comment (ZevEisenberg) [inline Pod/Classes/RZMessagingWindow.m]: I don't think the style guide says anything about using dot notation for pseudo-properties. Feel free to submit a PR if you think it should, but I'm not hugely bothered by it. I know some other people are, though.

Comment (jvisenti) [inline Pod/Classes/RZMessagingWindow.m]: Ah upon checking I see that the style only restricts use of dot syntax for non-idempotent methods.

**Pull request merged**